pkg/util/util.go

```diff
@@ -13,6 +15,7 @@ const ExtensionAPIServerAuthenticationConfigMap = "extension-apiserver-authentic
 const RequestHeaderClientCAFileKey = "requestheader-client-ca-file"
 const VirtShareDir = "/var/run/kubevirt"
 const VirtLibDir = "/var/lib/kubevirt"
+const GpuDevice = "nvidia.com/"
```

Comment:
I'm wondering why does it have to be company-specific?

Reply:
What comes to my mind is that if we make it agnostic, that we'll then have to somehow document what a device-plugin needs to fullfill in order to work with KubeVirt's passthrough implementation.

Reply:
Yes we can make the interaction with device plugin generic. To make it generic we will have to do the follow

1. As @fabiand mentioned define an interaction mechanism with device plugin. Any device plugin providing GPU passthrough devices will be bound by that interaction mechanism
2. Update the VMI object such that GPU device can be request in a generic way.

The current implementation of request GPU device is not very generic. It is as followa
There is no change in the Kubevirt API, we are leveraging the `DomainSpec.ResourceRequirements` to add a `limit` and `request` for GPU resource. 

The device plugin would expose GPUs with resource name of format `nvidia.com/<gpu_name>`

VM or VMI object can request GPU by adding `limit` and `request` in the format `nvidia.com/<gpu_name> : 1` in their `DomainSpec.ResourceRequirements` 

An example of VMI with GPU request is as below
```
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachineInstance
metadata:
  labels:
  name: vmi-gpu
spec:
  domain:
    devices:
      disks:
      - disk:
          bus: virtio
        name: containerdisk
    resources:
      limits:
        nvidia.com/<gpu_name>: 1
      requests:
        nvidia.com/<gpu_name>: 1
        memory: 1024M
  volumes:
  - containerDisk:
      image: registry:5000/kubevirt/fedora-cloud-container-disk-demo:devel
    name: containerdisk
```

If there is a consensus about making the interactions with device plugin generic I can help to design and build it.

Reply:
I'm all for making this generic. Same as we don't explicitly specify the vendor name when we assign SRIOV device we should keep GPU assignment generic as well.
 From my point of view, something like `nvidia.com/nvidia-11` would represent a resource name. Instead, I would expect something like `gpu/nvidia-11`

Reply:
The k8s device plugin specification demands that devices should be expose as `vendor-domain/resourcetype`. 
Instead of overloading the `DomainSpec.ResourceRequirements` is it a better idea to add a field in `DomainSpec.Devices`? The field can be called `gpus`. In this field user can specify the GPU vendor and type. The VMI yaml for requesting a GPU would look something like this 
```
apiVersion: kubevirt.io/v1alpha3
kind: VirtualMachineInstance
metadata:
  labels:
  name: vmi-gpu
spec:
  domain:
    devices:
      disks:
      - disk:
          bus: virtio
        name: containerdisk
      gpus:
      - gpu:
          name: nvidia.com/<gpu_name>
          count: 1
     - gpu:
          name: <some-vendor>/<some-gpu>
          count: 1
    resources:
      requests:
        memory: 1024M
  volumes:
  - containerDisk:
      image: registry:5000/kubevirt/fedora-cloud-container-disk-demo:devel
    name: containerdisk
```
As you have mentioned in the comment below the device plugin would set env variables  `GPU-PASSTHROUGH-DEVICES` and `VGPU-PASSTHROUGH-DEVICES`. The libvirt xml would be created based on which env variable is set.

I think this will make it generic and open for anyone else to implement their GPU device plugin. What do you think?